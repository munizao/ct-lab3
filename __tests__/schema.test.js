const {
  Schema
} = require('../lib/schema.js');

describe ('schema class module', () => {
  const testSchemaDefinition =
  {
    name: {
      type: String,
      required: true
    }, 
    age: {
      type: Number,
      required: true
    },
    weight: {
      type: String
    }
  };

  const goodDog = {
    name: 'frodo',
    age: 12,
    weight: '10 lbs'
  };

  const badDog = {
    name: 'bilbo',
    age: [1, 2, 3, 4]
  };

  const castingDog = {
    name: '',
    age: '12',
    weight: '20 lbs'
  };

  const castedDog = {
    name: '',
    age: 12,
    weight: '20 lbs'
  };

  const testSchema = new Schema(testSchemaDefinition);

  it('properly makes a schema', () => {
    expect(testSchema.validators).toBeDefined();
  });
  it('correctly validates a schema', () => {
    expect(testSchema.validate(goodDog)).toEqual(goodDog);
    expect(testSchema.validate(castingDog)).toEqual(castedDog);
    expect(() => testSchema.validate(badDog)).toThrowErrorMatchingSnapshot();
  });

});
