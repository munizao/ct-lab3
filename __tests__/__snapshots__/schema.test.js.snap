// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema class module correctly validates a schema 1`] = `"Cannot cast >>1,2,3,4<< to Number"`;
